src/Data/Primitive/Contiguous/Class.hs

```diff
@@ -884,6 +866,9 @@ instance Contiguous SmallArray where
   shrink !arr !n = do
     shrinkSmallMutableArray arr n
     pure arr
+  unsafeShrinkAndFreeze !arr !n = do
+    shrinkSmallMutableArray arr n
+    unsafeFreezeSmallArray arr
```

Comment:
This needs an `INLINE`, or (better is) just use the new default with `shrink` not `resize`.